### PR TITLE
 E-Commerce Application

### PR DESCRIPTION
**Pull request by Yashbanu**
- This application can perform CRUD operation such as **create , delete , update , display**
- **home.blade.php** is the root file which navigate to all other pages
- **resources/views** contain the blade file
- **route/web.php** contain routing logic to various pages
- **public folder** contains images and CSS file
- **app/http/controllers** contain logic to handle the incoming request
- **app/http/model** defines the database 
- **database/migration** contains the file that describe the schema of the database
- **database/seeders** contains the file that insert the default value in the database

Home Page

image

Insert Product

image

Delete Product

image

Update Product

image
image

Display Product

image

### CODE REVIEW

File: form/database/migrations/2024_01_16_122809_create_products_table.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+
+    public function up(): void  
+    {
+        Schema::create('products', function (Blueprint $table) {
+            $table->id();
```
Comment (Naveen1895) [inline form/database/migrations/2024_01_16_122809_create_products_table.php:16]: @Yashbanu Can we change to `increments()` or `BigIncrements()` before that know the difference and add whichever suits best.

File: form/database/migrations/2024_01_16_122809_create_products_table.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            $table->string('name');
+            $table->string('category');
```
Comment (Naveen1895) [inline form/database/migrations/2024_01_16_122809_create_products_table.php:18]: @Yashbanu If possible add comments() for each columns..

Comment (Naveen1895) [inline form/database/migrations/2024_01_16_122809_create_products_table.php:18]: @Yashbanu Can we add VARCHAR length ?

File: form/database/seeders/ProductCollection.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+use Illuminate\Support\Facades\DB;
+
+class ProductCollection extends Seeder
+{
+    /**
+     * Run the database seeds.
+     */
+    public function run(): void
+    {
+      DB::table("products")->insert(
```
Comment (Naveen1895) [inline form/database/seeders/ProductCollection.php:16]: Awesome.. This was a surprise.. so tried inserting sample data ?..

Comment (Yashbanu) [reply form/database/seeders/ProductCollection.php:16]: @Naveen1895 yeah using seeders

File: form/app/Http/Controllers/ProductController.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+     } 
+
+
+     //UpdateProduct
+    public function UpdateProduct(Request $req){
+        $data=Product::find($req->id);
+        return view("update_product",["data"=>$data]);
+     }
+
+
```
Comment (Naveen1895) [inline form/app/Http/Controllers/ProductController.php:44]: @Yashbanu Can we reduce the empty lines ?

File: form/app/Http/Controllers/ProductController.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+     //DeleteProduct
+    public function DeleteProduct(Request $req){
+        $data=Product::find($req->id);
+        $data->delete();
+        return redirect("display");
+     } 
+
+
+     //UpdateProduct
+    public function UpdateProduct(Request $req){
```
Comment (Naveen1895) [inline form/app/Http/Controllers/ProductController.php:39]: Functions should start with small letter.. like `updateProduct()`

File: form/app/Models/Product.php [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Product extends Model
+{
+    use HasFactory;
+    public $timestamps = false;
```
Comment (Naveen1895) [inline form/app/Models/Product.php:11]: @Yashbanu why $timestamps is FALSE.. ?

File: form/database/seeders/ProductCollection.php (hunk 3 above)
Comment (Naveen1895) [reply form/database/seeders/ProductCollection.php:16]: @Yashbanu Awesome.. I am happy.. :)

File: form/database/migrations/2024_01_16_122809_create_products_table.php (hunk 2 above)
Comment (Yashbanu) [reply form/database/migrations/2024_01_16_122809_create_products_table.php:18]: @Naveen1895 Yeah , I will add .

Comment (Yashbanu) [reply form/database/migrations/2024_01_16_122809_create_products_table.php:18]: @Naveen1895 Okey , I will add comment to it

File: form/database/migrations/2024_01_16_122809_create_products_table.php (hunk 1 above)
Comment (Yashbanu) [reply form/database/migrations/2024_01_16_122809_create_products_table.php:16]: @Naveen1895 sure

File: form/app/Models/Product.php (hunk 6 above)
Comment (Yashbanu) [reply form/app/Models/Product.php:11]: @Yashbanu I just tried with both true and false to see its results , I can change it.

File: form/app/Http/Controllers/ProductController.php (hunk 5 above)
Comment (Yashbanu) [reply form/app/Http/Controllers/ProductController.php:39]: @Naveen1895 okey

**Comment by Naveen1895** [comment]
Why merged ? :/

**Pull request merged**